### ISSUE TITLE
[DOC] Describe differences between Ansible group variables and group/node data

### ISSUE THREAD

**Issue by ipspace** [ipspace/netlab#1010]
### Describe the bug

When setting an Ansible variable in a custom group, Ansible inventory correctly inherits the changed setting while the `adjust_inventory_host` does not.

### To Reproduce

See #1007 for details.

### Expected behavior

`ansible-inventory` and `netlab connect` should get the same idea what **ansible_connection** variable value should be.

### Lab topology

```
provider: clab

groups:
  linux_ssh:
    members: [ h ]
    vars:
      ansible_connection: ssh

nodes:
  h:
    device: linux
```

### Version

All netlab versions

### Additional context

There are two ways to solve this challenge:

1. Consider custom groups and their **vars** settings in `adjust_inventory_host`
2. Do full inheritance before storing the data in the snapshot file

(1) might seem more complex, but we'd have to have the code to do the inheritance somewhere anyway, so maybe it doesn't matter that much. (2) would result in extra data in **netlab inspect** or **netlab create --output yaml**.

**Comment by sdktr** [comment]
This comment should probably have gone here:
https://github.com/ipspace/netlab/discussions/1007#discussioncomment-8098278

**Comment by ipspace** [comment]
After mulling this over a bit more: Ansible group variables within custom groups seems a fringe functionality, and I have no idea whether anyone is using them. It doesn't seem worth it to add another bit of potentially-brittle code to support that, so here's plan B: document what's going on and what a better way of handling the custom attributes is in https://netlab.tools/groups/#group-variables

Any thoughts/objections @sdktr @ssasso @jbemmel @petercrocker

**Comment by sdktr** [comment]
If you mean: let's not output ansible-groups anymore, I'm all for that. netlab has extensive logic to generate host specific variables. These can be put into the applicably ansible host vars directly. 

Why do we need ansible groups (custom or default ones) anyway given that we already know the host specific vars?

**Comment by ipspace** [comment]
> Why do we need ansible groups (custom or default ones) anyway given that we already know the host specific vars?

There are a few reasons I decided to use Ansible groups when designing _netlab_:

* There are fewer variables per host (I guess that doesn't matter any longer considering how big the data model is)
* You can change Ansible-related stuff in one place (group variable) without recreating Ansible inventory from the lab topology
* You can add any host- or group variables you wish -- that's why I'm using host/group directories instead of simple YAML files
* You can add stuff to Ansible groups that can later be used in templates without going through the netlab attribute definition process.

Unfortunately, I have no idea how people use _netlab_, but considering how creative networking engineers tend to get while trying to hammer a star-shaped peg into triangular hole, I wouldn't be surprised if someone built an ingenious solution relying on the above.

Long story short: I'll document the shortcomings and the differences between Ansible variables and **netlab connect**, and point to the new _change ansible connection method_ example.
